src/hooks/transactions/useTransactionType.ts

```diff
@@ -53,11 +56,14 @@ export default function useTransactionType({ input, type, weightInput }: Props):
     weight: null
   });
 
+  const planck = 10 ** targetApi.registry.chainDecimals[0];
+  const [resInp, msg] = evalUnits(new BN(input).mul(new BN(planck)).toString());
+  const transferInput = resInp?.toString() || '0';
```

Comment:
I think this logic can be included in the reducer and spawn the formatted value across all the dependent components. You would have to add to the `setTransferAmount` action the `targetApi.registry.chainDecimals[0]` so you can use it in the reducer through the action payload. Also there you could omit all your validations like `transferInput || "0"` in components call:

```
    case TransactionActionTypes.SET_TRANSFER_AMOUNT: {
       const {chainDecimals,input} = action.payload;
       const planck = 10 ** chainDecimals;
       const [resInp, msg] = evalUnits(new BN(input).mul(new BN(planck)).toString());
       const transferInput = resInp?.toString() || '0';
      return {
        ...state,
        transferAmount: transferInput,
        transferAmountError: action.payload.transferAmountError
      };
}
```

btw ... are you using `msg` that comes from `evalUnits` ?

Reply:
This logic cannot be included in the reducer as this will make alter the purpose of `case TransactionTypes.TRANSFER:` inside of useTransactionType and a need will be risen for it to be removed, thus decoupling the cases to more than one files.

Reply:
Sorry , i'm not following you ... could you please elaborate more the restriction for moving this logic to the reducer?

Reply:
Each case of the switch is dependent on the input that arrives in the useTransactionType.
Based on the type (Transfer, Call, Remark), weightInput and input (string) actions are made.
By altering the ones you mention above the `TransactionActionTypes.SET_TRANSFER_AMOUNT` actions are irrelevant to the incoming input and now solely depend to the transferAmount.

You probably have a point here - transferring the above inside the clever dispatcher - but I would prefer to implement that in a different issue, and keep this only for the bug fixing. Objections?

Reply:
No objections! is fine to restrict the scope so PR does not get huge.

I think this can be included in https://github.com/paritytech/parity-bridges-ui/issues/195